code/modules/admin/verbs/maprotation.dm

```diff
@@ -68,28 +65,57 @@
 			return
 
 		qdel(M)
+		var/config_file = null
+		var/list/json_value = list()
+		var/config = tgui_alert(usr,"Would you like to upload an additional config for this map?", "Map Config", list("Yes", "No"))
+		if(config == "Yes")
+			config_file = input("Pick file:", "Config JSON File") as null|file
+			if(isnull(config_file))
+				return
+			if(copytext("[config_file]", -5) != ".json")
+				to_chat(src, span_warning("Filename must end in '.json': [config_file]"))
+				return
+			if(fexists("data/custom_map_json/[config_file]"))
+				fdel("data/custom_map_json/[config_file]")
+			if(!fcopy(config_file, "data/custom_map_json/[config_file]"))
+				return
+			if (VM.LoadConfig("data/custom_map_json/[config_file]", TRUE) != TRUE)
```

Comment:
as youre touching this proc, please change all the nondescript variable names to readable ones